### ISSUE TITLE
'WithMessage'  not work for custom asynchronous validators.

### ISSUE THREAD

**Issue by Maxim-Chugaev** [ctavan/express-validator#203]
Asynchronous validator
```
Channel.channelNameAvailable = (name, id = 0)->
  Channel
    .query({'where':{'name': name}, 'whereNot': {'id': id}})
    .fetch()
    .then (channel)->
      Promise.reject('channel name already in use') if channel?
```
Initialize express validator
```
router.use expressValidator
                  customValidators:
                    channelNameAvailable: Channel.channelNameAvailable
```
Validation

```
req
        .check('name')
        .notEmpty().withMessage('channel name is required')
        .isLength({min: 4, max: 20}).withMessage('channel name must have length in range between 4 and 20')
        .channelNameAvailable(req.body.id).withMessage('channel name already in use')
```

if channel name is already exists, i get error 'Invalid value', but must be another message!

**Comment by LordWingZero** [comment]
I am also having this issue. Came here to post and saw this.

Please fix.

**Comment by rustybailey** [comment]
@Jakeii or @chrissinclair -- Would one of you guys be able to help with this?

**Comment by chrissinclair** [comment]
I'll see if I can't take a look over the weekend. @Maxim-Chugaev @LordWingZero, do either of you have a minimal example showing this already I can use as a base?

**Comment by LordWingZero** [comment]
# Bare-bones test case showcasing error:

## app.js
```Javascript
var express = require('express');
var bodyParser = require('body-parser');
var expressValidator = require('express-validator');
var Promise = require("bluebird");

var app = express();

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({
    extended: true
}));
app.use(expressValidator({
    customValidators: {
        testValidator: function(username) {
            return new Promise(function(resolve, reject) {
                reject();
            });
        }
    }
}));

app.get('/', function (req, res) {
    res.send([
        '<form method="post">',
        '<input name="testInput" />',
        '</form>',
        ].join(''));
});

app.post('/', function(req, res) {
    
    req.checkBody('testInput', 'Default Message')
        .testValidator()
        .withMessage('With Message');

    req.asyncValidationErrors()
    .then(function(){
        return res.send('Validation succeeded.');        
    })
    .catch(function(errors) {
        return res.send(errors);
    });

});

app.listen(process.env.PORT, process.env.IP, function() {
    console.log('Example app listening on port 3000!');
});
```

## package.json
```JSON
{
  "name": "express-validator-test-case",
  "version": "0.0.0",
  "description": "A test case showing the asyncCustomValidationErrors",
  "main": "app.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "bluebird": "^3.3.3",
    "body-parser": "^1.15.0",
    "express": "^4.13.4",
    "express-validator": "^2.19.1"
  }
}

```

**Comment by chrissinclair** [comment]
@LordWingZero Thanks for that - I can replicate the issue, and have an approach that's looking promising. I'll need to spend a bit more time tomorrow / sunday to make sure that it's robust, will submit the PR once it is.

**Comment by rustybailey** [comment]
Thanks for looking into this so quickly, @chrissinclair!

**Comment by chrissinclair** [comment]
I've just submitted the PR for this - let me know your thoughts :)

**Comment by rustybailey** [comment]
This fix is now in v2.19.2.
